parsing/parser.mly

```diff
@@ -418,6 +418,11 @@ let wrap_sig_ext ~loc body ext =
 let wrap_mksig_ext ~loc (item, ext) =
   wrap_sig_ext ~loc (mksig ~loc item) ext
 
+let mk_quotedext ~loc (id, idloc, str, strloc, delim) =
+  let exp_id = mkloc id idloc in
+  let e = ghexp ~loc (Pexp_constant (Pconst_string (str, strloc, delim))) in
```

Comment:
Why is this a `ghexp`? If I understand correctly, this corresponds to a single token in the stream with a well-defined global location, so why make it ghost?

(Also, maybe this helper function could be defined with the other expression-creation functions above, for example after `exp_of_longident`, rather than between wrapper functions and documentation-comments functions?)

Reply:
I've never been quite sure what is considered ghost and what is not. In this case, there is no substring in the input that form the syntax for quoted string, so I marked it as ghost. If that's not the criterion, then I would like someone to document what it is. :)

Reply:
Summon @trefis, our expert in ghosts.

You have a point, but I would like to ensure that either the `_extension` AST node or the `String` AST node is non-ghost; I wasn't able to convince myself that the extension would be non-ghost, so maybe one should think of these tokens as "clearly a quoted string, with a ghostly whiff of extension-point"?

Reply:
The extension itself (`Pexp_extension`/...) is non-ghost. See the tests for convincing.

Reply:
Great! Thanks.